test/server/hot_restart_impl_test.cc

```diff
@@ -0,0 +1,47 @@
+#include "server/hot_restart_impl.h"
+
+#include "test/mocks/server/mocks.h"
+
+#include "gtest/gtest.h"
+
+using testing::Invoke;
+using testing::InvokeWithoutArgs;
+using testing::Return;
+using testing::WithArg;
+using testing::_;
+
+namespace Envoy {
+namespace Server {
+
+TEST(HotRestartImplTest, alloc) {
+  MockOsSysCalls os_sys_calls;
+  NiceMock<MockOptions> options;
+  CSmartPtr<void, ::free> buffer;
```

Comment:
nit: for simplicity I would probably just make this a vector<uint8_t> or something like that (the you can just call resize(), return address of [0], etc.). But don't feel that strongly about it.

Reply:
Good idea.  Side note: c++11 added std::vector::data() so we don't have to take the address of element 0 anymore: http://en.cppreference.com/w/cpp/container/vector/data